pypsa/component_attrs/generators.csv

```diff
@@ -34,6 +34,7 @@ weight,float,n/a,1,"Weighting of a generator. Only used for network clustering."
 p,series,MW,0,active power at bus (positive if net generation),Output
 q,series,MVar,0,reactive power (positive if net generation),Output
 p_nom_opt,float,MW,0,Optimised nominal power.,Output
+n_opt,int,n/a,0,Optimised number of generators.,Output
```

Comment:
As discussed the optimized variable shall be named "n_mod_opt" as output.
Morever, we need a new input parameter "p_nom_mod" that is the size of each module.
The logic shall be equivalent with the one we proposed: "if 0, we preserve the continuous formulation, if > 0, we implement modularity"